### DISCUSSION TITLE
Local daily forecasts starting at midnight?

### DISCUSSION THREAD

**Discussion by slominskir** [weather-gov/api#695] [Q&A]
I'm looking for a query to obtain local forecast data organized daily starting and ending each day at midnight.  A 5 day forecast at this frequency would be sufficient.   Is this possible with this API or any weather.gov API?   I believe this is about the most commonly seen forecast these days, yet I seem to be overlooking how to query NWS for it.   I can obtain a 12 hour forecast starting at 6am/6pm local time or an hourly forecast, but not a 24 hour forecast starting at midnight.  Please point me in the right direction with an example if possible.  Thanks.

**Comment by ed5464215** [comment]
What I do is use two queries: one for the period nearest to the present, and one for +8 hours in the future.

**Reply by slominskir** [reply]
Sounds like you're suggesting merging 12 hour forecasts together to create a 24 hour forecast.  These forecasts won't start at midnight (at least in my local zone).  Plus, doing the merge client-side sounds problematic at best (I'm looking for single condition phrase and icon per day).  I did stumble on the older [ndfdBrowserClientByDay.php](https://graphical.weather.gov/xml/sample_products/browser_interface/ndfdBrowserClientByDay.php?whichClient=NDFDgenByDay&lat=37.08&lon=-76.51&format=24+hourly&startDate=2024-01-02&numDays=7&Unit=e) which appears to do this for you server-side, but again, starts at 6am daily, not midnight (and more problematically the precipitation field is still tied to a separate 12-hour time series).  I haven't looked closely at how it is determining the condition and icon for the 24 hour period, and it may be doing something simple (and disappointing) like simply taking the daytime 12 hour forecast and ignoring the overnight.

As far as I can tell the most common forecasts are hourly and daily, and therefore the most common user expectation is that an American can pull a phone out of their pocket and whichever weather app they're using likely provides this (along with daily highs/lows, daily precipitation probability, a daily conditions phrase and icon, and sunrise/sunset, etc).  Oddly it appears NWS can't easily provide this same weather info with a simple query.

**Comment by mikeiz404** [comment]
I'm trying to figure this out as well. For the weather condition, one approach may be to take the most severe weather condition from the hourly forecasts you are interested in.
[open-meteo.com](https://open-meteo.com/en/docs) does something like this (see "WMO Weather interpretation codes" and "weather_code"). Their weather code is a number so I'm assuming they take the max over the hourly weather codes or something along those lines. For weather.gov you could do something similar but you would need to know the correct ranking of the weather conditions.

A few other relevant links:
- https://github.com/weather-gov/api/discussions/659
- https://www.nodc.noaa.gov/archive/arc0021/0002199/1.1/data/0-data/HTML/WMO-CODE/WMO4677.HTM
- https://api.weather.gov/icons
- https://www.weather.gov/forecast-icons/
